canela/fingerprints/soap.py

```diff
@@ -0,0 +1,86 @@
+import time
+
+import click
+from dscribe.descriptors import SOAP
+
+from canela.io.cube import read_cube
+from canela.io.pickle import write_pickle
+from canela.fingerprints.grid import calc_grid
+
+
+path = './'
+file_cube = 'aiida-rho-ELECTRON_DENSITY-1_0.cube'
+
+print('Reading cube...')
+start = time.time()
+cube = read_cube(path + file_cube)
+elapsed = time.time() - start
+print('DONE in ', elapsed, '\n')
+
+print('Calculating grid...')
+start = time.time()
+grid = calc_grid(cube)
+elapsed = time.time() - start
+print('DONE in ', elapsed, '\n')
+print('Shape of grid:', grid.shape)
+
+struc = cube['atoms']
+rho = cube['data']
+
+print('Writing grid to disk...')
+start = time.time()
+write_pickle(grid, path + 'grid')
+elapsed = time.time() - start
+print('DONE in ', elapsed, '\n')
+
+print('Flattening rho...')
+start = time.time()
+rho = rho.flatten(order='C')  # flatten in row-major order to match grid order
+rho = rho.reshape(-1, 1)
+elapsed = time.time() - start
+print('DONE in ', elapsed, '\n')
+print('Shape of rho:', rho.shape)
+
+print('Writing rho to disk...')
+start = time.time()
+write_pickle(rho, path + 'rho')
+elapsed = time.time() - start
+print('DONE in ', elapsed, '\n')
+
```

Comment:
All the above should be in your click command! All you need to do is add an argument for filepath.

Reply:
It appears that soap.py has two separate scripts. I'm going to talk to Dylan before making any changes, but I will eventually move the scripts over to canela.bin